schemas/layer-4.cue

```diff
@@ -1,37 +1,108 @@
 package schemas
 
 #Layer4: {
-    evaluations: [#ControlEvaluation, ...#ControlEvaluation]
+	evaluations: [#ControlEvaluation, ...#ControlEvaluation]
 }
 
-// Types
-
+// ControlEvaluation provides all assessment results for a single control
 #ControlEvaluation: {
-    name: string
-    "control-id": string
-    result: #Result
-    message: string
-    "documentation-url"?: =~"^https?://[^\\s]+$"
-    "corrupted-state"?: bool
-    "assessment-results"?: [...#AssessmentResult]
+	// Name is the name of the control being evaluated
+	name: string
+	// ControlID uniquely identifies the control
+	"control-id": string
+	// Result communicates whether the evaluation has been run, and if so, the outcome(s)
+	result: #Result
+	// Message describes the result of the evaluation
+	message: string
+	// CorruptedState is true if the control evaluation was interrupted and changes were not reverted
+	"corrupted-state"?: bool
+	// RemediationGuide provides a URL with guidance on how to remediate systems that fail control evaluation
+	"remediation-guide"?: string
+	// Assessments represents the collection of results from evaluation of each control requirement
+	assessments: [...#Assessment]
+}
+
+// Assessment provides all assessment results from evaluation of a single control requirement
+#Assessment: {
+	// RequirementID uniquely identifies the requirement being tested
+	"requirement-id": string @go(RequirementId)
+	// Applicability provides identifier strings to determine when this assessment is applicable
+	applicability: [...string]
+	// Description provides a detailed explanation of the assessment
+	description: string
+	// Result communicates whether the assessment has been run, and if so, the outcome(s)
+	result: #Result
+	// Message describes the result of the assessment
+	message: string
+	// Procedures defines the assessment procedures associated with the assessment
+	procedures: [...#AssessmentProcedure]
+	// RunDuration is the time it took to run the assessment
+	"run-duration"?: string @go(RunDuration)
```

Comment:
Why do the typical data consumers care how long the assessment took? Or, put another way, what value does this data provide and to whome?

Reply:
Agree. This was part of the attempt to align the schema with the current struct in the Go library. I would proposed start and end timestamps instead.